lib/auth/auth_login_test.go

```diff
@@ -1083,6 +1079,94 @@ func TestServer_Authenticate_passwordless(t *testing.T) {
 	}
 }
 
+// TestServer_Authenticate_passwordlessSSO tests a scenario where an SSO user
+// bypasses SSO by using a passwordless login.
+func TestServer_Authenticate_passwordlessSSO(t *testing.T) {
```

Comment:
You're free to continue to disagree with me, but I will continue to express my distaste for mixing upper case, lower case, and underscores in a function name.

```suggestion
func TestPasswordlessProhibitedForSSO(t *testing.T) {
```

Reply:
Done, here and at lib/web.